src/main/java/lotto/domain/Result.java

```diff
@@ -18,11 +18,9 @@ public Map<Rank, Integer> result() {
         return result;
     }
 
-    public void update(int matchCount) {
-        if (3 <= matchCount && matchCount <= 6) {
-            result.replace(Rank.rankByMatchCount(matchCount),
-                result.get(Rank.rankByMatchCount(matchCount)) + 1);
-        }
+    public void update(Rank rank) {
+        result.replace(rank,
+            result.get(rank) + 1);
```

Comment:
중요한 부분은 아니나 줄바꿈이 일반적인 형태는 아닌 것 같아요. 한줄로 작성하거나 아래처럼 작성하는건 어떨까요?

```suggestion
        result.replace(
            rank,
            result.get(rank) + 1
        );
```